ros_ws/src/decision/src/thomas_tracker.cpp

```diff
@@ -0,0 +1,232 @@
+/** \file thomas_tracker.cpp
+ * \brief Simple targeting node
+ *
+ * \author Sébastien Darche <sebastien.darche@polymtl.ca>
+ * \modifications Thomas Petrie <thomas.petrie@polymtl.ca> 
+ */
+
+// Std includes
+
+#include <algorithm>
+
+// ROS includes
+
+#include <ros/ros.h>
+
+#include "serial/Target.h"
+#include "tracking/Tracklets.h"
+
+struct BoundingBox {
+    float upper_edge;
+    float lower_edge;
+    float left_edge;
+    float right_edge;
+
+    float score = 0.f;
+
+    int clss;
+
+    BoundingBox(tracking::Tracklet& bbox) {
+        upper_edge = bbox.y + bbox.h / 2.f;
+        lower_edge = bbox.y - bbox.h / 2.f;
+        left_edge = bbox.x - bbox.w / 2.f;
+        right_edge = bbox.x + bbox.w / 2.f;
+        clss = bbox.clss;
+    }
+
+    bool contains(BoundingBox& bebe) {
+        return (this->upper_edge > bebe.upper_edge && this->lower_edge < bebe.lower_edge&& this->left_edge  < bebe.left_edge&& this->right_edge > bebe.right_edge);
+    } // à updater avec sébastien
+};
+
+
+class SimpleTracker {
+
+
+  public:
+    SimpleTracker(ros::NodeHandle& n, int _enemy_color)
+        : nh(n), enemy_color(_enemy_color) {
+        sub_tracklets = nh.subscribe("tracklets", 1,
+                                     &SimpleTracker::callbackTracklets, this);
+
+        pub_target = nh.advertise<serial::Target>("target", 1);
+        std::cout << "Enemy color set to be: "
+                  << (enemy_color == 0 ? "red" : "blue") << "\n";
+    }
+
+
+    void callbackTracklets(const tracking::TrackletsConstPtr& trks) {
+        auto distance = [](auto d1, auto d2) {
+            return std::sqrt(std::pow(d1.x - d2.x, 2) + std::pow(d1.y - d2.y, 2));
+        };
+
+        /*float best_dist = INFINITY;
+        int index = -1;
+
+        int i = 0;
+
+        for (auto trk : trks->tracklets) {
+            auto dist = distance(last_trk, trk);
+            if (dist < best_dist && enemy_color == int(trk.clss)) {
+                index = i;
+                best_dist = dist;
+            }
+            ++i;
+        } 
+
+        if (index != -1) {
+            last_trk = trks->tracklets[index];
+            pub_target.publish(toTarget(last_trk));
+        }code de sébastien*/
+
+        auto boxSize = [](auto bbox) {
+            return bbox.w * bbox.h;
+        };
+
+        auto findBoxes = [&](auto bbox) {
```

Comment:
Why not create another function instead of nesting it here ?

Reply:
Move to BoundingBox struct/class, make it a function